stellargraph/layer/gcn.py

```diff
@@ -181,30 +187,33 @@ def call(self, inputs):
             Keras Tensor that represents the output of the layer.
         """
         features, *As = inputs
-        batch_dim, n_nodes, _ = K.int_shape(features)
-        if batch_dim != 1:
-            raise ValueError(
-                "Currently full-batch methods only support a batch dimension of one"
-            )
-
-        # Remove singleton batch dimension
-        features = K.squeeze(features, 0)
 
         # Calculate the layer operation of GCN
         A = As[0]
-        h_graph = K.dot(A, features)
+        if K.is_sparse(A):
+            # batch_dot doesn't support sparse tensors, so we special case them to only work with a
+            # single batch element (and the adjacency matrix without a batch dimension)
```

Comment:
So is this something we don't need to support? Or is it something to fix?

Reply:
Hmm I see 🤔 

Ah by "is this something we don't need to support" I actually meant do we know if users will run into these errors (in which case, maybe suggesting to use dense matrices would be better than suggesting that the batch dimension in wrong?), but sounds like they shouldn't if they are using the generators we provide.

Reply:
Yeah, they shouldn't hit it with our generators. If the generators are behaving incorrectly, hopefully that's something we'd catch first.

(Also, it turns out a sparse batch dot may not be _that_ hard: https://github.com/tensorflow/tensorflow/issues/9210#issuecomment-497889961)

I'll file an issue.